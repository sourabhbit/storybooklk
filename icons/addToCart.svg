<svg width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.2468 11.7257H7.35978L4.48144 0.239258H0.32666V1.65844H3.37161L6.24996 13.1449H15.1926L18.9731 4.09277L17.6608 3.54557L14.2468 11.7257Z" fill="black"/>
<path d="M6.80349 18.1798C7.59903 18.1798 8.24394 17.5364 8.24394 16.7427C8.24394 15.9491 7.59903 15.3057 6.80349 15.3057C6.00795 15.3057 5.36304 15.9491 5.36304 16.7427C5.36304 17.5364 6.00795 18.1798 6.80349 18.1798Z" fill="black"/>
<path d="M14.0008 18.1798C14.7963 18.1798 15.4412 17.5364 15.4412 16.7427C15.4412 15.9491 14.7963 15.3057 14.0008 15.3057C13.2052 15.3057 12.5603 15.9491 12.5603 16.7427C12.5603 17.5364 13.2052 18.1798 14.0008 18.1798Z" fill="black"/>
<path d="M11.8324 7.4095V4.88822H14.3596V3.46905H11.8324V0.947754H10.4099V3.46905H7.88263V4.88822H10.4099V7.4095H11.8324Z" fill="black"/>
</svg>
