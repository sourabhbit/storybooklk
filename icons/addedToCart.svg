<svg width="20" height="19" viewBox="0 0 20 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.461 11.7257H7.57658L4.69823 0.239258H0.543457V1.65844H3.58584L6.46421 13.1449H15.4094L19.1899 4.09277L17.8776 3.54557L14.461 11.7257Z" fill="#000042"/>
<path d="M8.44037 16.8823C8.51873 16.0925 7.9405 15.3889 7.14884 15.3107C6.35718 15.2325 5.65189 15.8094 5.57353 16.5992C5.49517 17.389 6.0734 18.0927 6.86506 18.1709C7.65672 18.249 8.36201 17.6721 8.44037 16.8823Z" fill="#000042"/>
<path d="M15.6362 16.8814C15.7146 16.0916 15.1363 15.388 14.3447 15.3098C13.5531 15.2316 12.8478 15.8085 12.7694 16.5983C12.6911 17.3881 13.2693 18.0917 14.0609 18.1699C14.8526 18.248 15.5578 17.6712 15.6362 16.8814Z" fill="#000042"/>
<path d="M14.2996 1.00391L10.8856 4.97249L8.92479 3.28226L7.99438 4.35624L11.0342 6.97469L15.3786 1.92701L14.2996 1.00391Z" fill="#000042"/>
</svg>
