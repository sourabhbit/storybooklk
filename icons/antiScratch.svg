<svg width="20" height="15" viewBox="0 0 20 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.548096 14.6468H19.2407V0.306641H0.548096V14.6468ZM17.8182 13.2276H14.2119L17.8182 9.45078V13.2276ZM17.8182 7.39231L12.2486 13.2276H8.35271L17.8182 3.31377V7.39231ZM17.3697 1.72326L6.38938 13.2276H2.49348L13.4763 1.72326H17.3697ZM1.96805 1.72326H5.65121L1.96805 5.58191V1.72326ZM1.96805 7.6378L7.61454 1.72326H11.5104L1.96805 11.7189V7.6378Z" fill="#000042"/>
</svg>
