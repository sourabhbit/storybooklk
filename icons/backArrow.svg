<svg width="20" height="16" viewBox="0 0 20 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M9.08475 1.53069L8.08002 0.52832L0.383057 8.20724L0.431756 8.25838L8.08002 15.8887L9.08475 14.8864L3.1025 8.91812H19.0911V7.49894H3.1025L9.08475 1.53069Z" fill="#000042"/>
</svg>
