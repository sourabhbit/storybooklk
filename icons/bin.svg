<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M15.5385 0.883789H4.04302V3.75539H0.436768V5.17457H2.60513V19.5326H16.9789V5.17457H19.1473V3.75539H15.541V0.883789H15.5385ZM5.46299 2.30297H14.116V3.75539H5.46299V2.30297ZM15.5564 5.17457V18.1134H4.0251V5.17457H4.04302H15.5385H15.5564Z" fill="#000042"/>
<path d="M6.90347 6.61914H5.48096V16.6711H6.90347V6.61914Z" fill="#000042"/>
<path d="M10.502 6.61914H9.07947V16.6711H10.502V6.61914Z" fill="#000042"/>
<path d="M14.1007 6.61914H12.6782V16.6711H14.1007V6.61914Z" fill="#000042"/>
</svg>
