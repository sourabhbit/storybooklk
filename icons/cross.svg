<svg width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1.06393 18.5925L9.19661 10.4788L17.3267 18.5925L18.334 17.5875L10.2013 9.4739L18.334 1.36284L17.3267 0.35791L9.19661 8.47153L1.06393 0.35791L0.0566406 1.36284L8.18932 9.4739L0.0566406 17.5875L1.06393 18.5925Z" fill="#000042"/>
</svg>
