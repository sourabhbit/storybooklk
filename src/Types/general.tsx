export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
