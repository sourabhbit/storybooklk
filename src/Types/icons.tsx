export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
