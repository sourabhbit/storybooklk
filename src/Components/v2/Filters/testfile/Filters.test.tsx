import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
import FiltersComp from '../index'

const props = {
	id: 'active-id',
	text: 'Active Filters',
	count: 10,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.primary,
	font: TypographyENUM.serif,
	showIcon: false,
	icon: iconsENUM.glasses,
	onClick: jest.fn(),
}

describe('<FiltersComp />', () => {
	let _asFragment: any = null
	let _rerender: any = null

	beforeEach(() => {
		const { asFragment, rerender } = render(<FiltersComp {...props} />)
		_asFragment = asFragment
		_rerender = rerender
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('filter-wrapper')
		const count = screen.getByTestId('filter-count')
		const text = screen.getByTestId('filter-text')
		// const color = screen.getByTestId('filter-color')
		expect(container).toBeInTheDocument()
		expect(count).toBeInTheDocument()
		expect(text).toBeInTheDocument()
		// expect(color).toBeInTheDocument()
	})

	it('Should not render component', () => {
		_rerender(<FiltersComp {...props} count={undefined} />)
		const count = screen.queryByTestId('filter-count')
		const color = screen.queryByTestId('filter-color')
		expect(count).not.toBeInTheDocument()
		expect(color).not.toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
