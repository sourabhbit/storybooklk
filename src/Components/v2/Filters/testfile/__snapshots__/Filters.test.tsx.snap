// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltersComp /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL fVXOHo"
    data-testid="filter-wrapper"
    id="active-id"
  >
    <span
      class="sc-dkPtRN juyXTR"
      data-testid="filter-text"
    >
      Active Filters
    </span>
    <span
      class="sc-gsDKAQ eSbkki"
      data-testid="filter-count"
    >
      10
    </span>
  </div>
</DocumentFragment>
`;
